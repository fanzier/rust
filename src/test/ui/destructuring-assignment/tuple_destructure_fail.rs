#![feature(destructuring_assignment)]

<<<<<<< HEAD
const C: i32 = 1;

=======
>>>>>>> Implement destructuring assignment (squashed)
fn main() {
    let (mut a, mut b);
    (a, .., b, ..) = (0, 1); //~ ERROR `..` can only be used once per tuple pattern
    (a, a, b) = (1, 2); //~ ERROR mismatched types
<<<<<<< HEAD
    (C, ..) = (0,1); //~ ERROR invalid left-hand side of assignment
=======
>>>>>>> Implement destructuring assignment (squashed)
    (_,) = (1, 2); //~ ERROR mismatched types
}
