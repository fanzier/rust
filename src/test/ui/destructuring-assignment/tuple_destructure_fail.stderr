error: `..` can only be used once per tuple pattern
<<<<<<< HEAD
  --> $DIR/tuple_destructure_fail.rs:7:16
=======
  --> $DIR/tuple_destructure_fail.rs:5:16
>>>>>>> Implement destructuring assignment (squashed)
   |
LL |     (a, .., b, ..) = (0, 1);
   |         --     ^^ can only be used once per tuple pattern
   |         |
   |         previously used here

error[E0308]: mismatched types
<<<<<<< HEAD
  --> $DIR/tuple_destructure_fail.rs:8:5
=======
  --> $DIR/tuple_destructure_fail.rs:6:5
>>>>>>> Implement destructuring assignment (squashed)
   |
LL |     (a, a, b) = (1, 2);
   |     ^^^^^^^^^   ------ this expression has type `({integer}, {integer})`
   |     |
   |     expected a tuple with 2 elements, found one with 3 elements
   |
   = note: expected type `({integer}, {integer})`
             found tuple `(_, _, _)`

<<<<<<< HEAD
error[E0070]: invalid left-hand side of assignment
  --> $DIR/tuple_destructure_fail.rs:9:13
   |
LL |     (C, ..) = (0,1);
   |      -      ^
   |      |
   |      cannot assign to this expression

error[E0308]: mismatched types
  --> $DIR/tuple_destructure_fail.rs:10:5
=======
error[E0308]: mismatched types
  --> $DIR/tuple_destructure_fail.rs:7:5
>>>>>>> Implement destructuring assignment (squashed)
   |
LL |     (_,) = (1, 2);
   |     ^^^^   ------ this expression has type `({integer}, {integer})`
   |     |
   |     expected a tuple with 2 elements, found one with 1 element
   |
   = note: expected type `({integer}, {integer})`
             found tuple `(_,)`

<<<<<<< HEAD
error: aborting due to 4 previous errors

Some errors have detailed explanations: E0070, E0308.
For more information about an error, try `rustc --explain E0070`.
=======
error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.
>>>>>>> Implement destructuring assignment (squashed)
